### PR TITLE
Add API Tokens to User for use with the API

### PR DESCRIPTION
**Pull request by jules2689**
What this does
---
- This adds an api token for use with the JSON API to all users
- It will not generate tokens for existing users, but will instead ask them to click the regenerate button (saves a migration)

cc @chrisarcand @tarebyte for review

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -36,8 +36,12 @@ def edit
   #
   def update
     if current_user.update_attributes(update_user_params)
+      if params[:user][:regenerate_api_token]
+        current_user.regenerate_api_token
+      end
+
       respond_to do |format|
-        format.html { redirect_to root_path }
+        format.html { redirect_back(fallback_location: root_path) }
```
Comment (jules2689) [inline app/controllers/users_controller.rb]: When you click "regenerate token" you probably want to be taken back... I think this is fine, but I'm not sure where else the user settings page is?

Comment (chrisarcand) [reply app/controllers/users_controller.rb]: No where else than `/settings`

Comment (jules2689) [reply app/controllers/users_controller.rb]: Think it makes sense to just redirect to settings?

**Comment by jules2689** [comment]
Here is the UI ^^

**Comment by chrisarcand** [comment]
Does it make sense to only have an option to regenerate and not reshow a token behind the first time you generate it? (ala GitHub's personal access tokens). 

Granted, the data we deal with here isn't exactly super sensitive (maybe?) and this might be nitpicking but it was a thought I immediately had. Thoughts? @tarebyte ?

**Comment by jules2689** [comment]
I'm fairly indifferent, but I think that the one time show would be a bit more complex as we'd have to include some flag stating that it was shown or not.

**Pull request merged**